src/node_api.cc

```diff
@@ -557,65 +576,67 @@ void napi_module_register(napi_module* mod) {
   node::node_module_register(nm);
 }
 
-#define RETURN_STATUS_IF_FALSE(condition, status)                       \
+#define RETURN_STATUS_IF_FALSE(env, condition, status)                  \
   do {                                                                  \
     if (!(condition)) {                                                 \
-      return napi_set_last_error((status));                             \
+      return napi_set_last_error((env), (status));                      \
     }                                                                   \
   } while (0)
 
-#define CHECK_ARG(arg) RETURN_STATUS_IF_FALSE((arg), napi_invalid_arg)
+#define CHECK_ENV(env) \
+  assert((env) != nullptr)
```

Comment:
Certainly it's nicer than segfaulting. But I was suggesting returning `napi_invalid_arg` instead, as in @gabrielschulhof's previous revision of this PR. As explained in my other comment:

> There is no need to bring down the whole process because there is a bug in some part of some native module's code. Aborts should be reserved for conditions when the node engine or JS VM cannot continue to execute reliably.

> When using the C++ wrapper the return codes are always checked and will propagate up as a JavaScript exception if unhandled.